client/asset/btc/btc.go

```diff
@@ -1859,6 +1866,472 @@ func (btc *baseWallet) fundedTx(coins asset.Coins) (*wire.MsgTx, uint64, []outPo
+	if fundsRequired > changeOutput.value {
+		// If change not enough, need to use other UTXOs.
+		utxos, _, _, err := btc.spendableUTXOs(1)
+		if err != nil {
+			return makeError(err)
+		}
+
+		_, _, additionalInputs, _, _, _, err = fund(utxos, func(inputSize, inputsVal uint64) bool {
+			txSize := dexbtc.MinimumTxOverhead + inputSize
+
+			// input is the change input that we must use
+			if btc.segwit {
+				txSize += dexbtc.RedeemP2PWKHInputTotalSize
+			} else {
+				txSize += dexbtc.RedeemP2PKHInputSize
+			}
+
+			if requiredForRemainingSwaps > 0 {
+				if btc.segwit {
+					txSize += dexbtc.P2WPKHOutputSize
+				} else {
+					txSize += dexbtc.P2PKHOutputSize
+				}
+			}
+
+			totalFees := additionalFeesRequired + txSize*newFeeRate
+			return totalFees+requiredForRemainingSwaps <= inputsVal+changeOutput.value
+		})
+		if err != nil {
+			return makeError(fmt.Errorf("failed to fund acceleration tx: %w", err))
+		}
+	}
+
+	baseTx, totalIn, _, err := btc.fundedTx(append(additionalInputs, changeOutput))
+	if err != nil {
+		return makeError(err)
+	}
+
+	changeAddr, err := btc.node.changeAddress()
+	if err != nil {
+		return makeError(fmt.Errorf("error creating change address: %w", err))
+	}
+
+	tx, output, txFee, err := btc.signTxAndAddChange(baseTx, changeAddr, totalIn, additionalFeesRequired, newFeeRate)
+	if err != nil {
+		return makeError(err)
+	}
+
+	return tx, output, txFee + additionalFeesRequired, nil
+}
+
+// lookupOutput looks up the value of a transaction output and creates an
+// output.
+func (btc *baseWallet) lookupOutput(txHash *chainhash.Hash, vout uint32) (*output, error) {
+	getTxResult, err := btc.node.getWalletTransaction(txHash)
+	if err != nil {
+		return nil, err
+	}
+	tx, err := msgTxFromBytes(getTxResult.Hex)
+	if err != nil {
+		return nil, err
+	}
+	if len(tx.TxOut) <= int(vout) {
+		return nil, fmt.Errorf("txId %s only has %d outputs. tried to access index %d",
+			txHash, len(tx.TxOut), vout)
+	}
+
+	value := tx.TxOut[vout].Value
+	return newOutput(txHash, vout, uint64(value)), nil
+}
+
+// changeCanBeAccelerated returns whether or not the change output can be
+// accelerated.
+func (btc *baseWallet) changeCanBeAccelerated(changeTxHash *chainhash.Hash, changeVout uint32, requiredForRemainingSwaps uint64) error {
+	lockedUtxos, err := btc.node.listLockUnspent()
+	if err != nil {
+		return err
+	}
+
+	var changeIsLocked bool
+	for _, utxo := range lockedUtxos {
+		if utxo.TxID == changeTxHash.String() && utxo.Vout == changeVout {
+			changeIsLocked = true
+			break
+		}
+	}
+
+	if changeIsLocked && requiredForRemainingSwaps == 0 {
+		return errors.New("change cannot be accelerated because it is locked by another order")
+	}
+
+	if !changeIsLocked {
+		utxos, err := btc.node.listUnspent()
+		if err != nil {
+			return err
+		}
+
+		var changeIsUnspent bool
+		for _, utxo := range utxos {
+			if utxo.TxID == changeTxHash.String() && utxo.Vout == changeVout {
+				changeIsUnspent = true
```

Comment:
Strikes me as a bit brute force, when `getTxOut` would ideally work.  listunspent is probably most straightforward however since `getTxOut` in spv might mess up and start a cfilters scan, which would suck.  No need to change it I guess.